### ISSUE TITLE
Logs

### ISSUE THREAD

**Issue by iamkarlson** [BladeMight/Mahou#39]
Your logging system is working less stable than [third party libraries](https://github.com/BladeMight/Mahou/pull/38)

```
10:25:11.968 [Error]
                    Unexpected error occured, Mahou exitted, error details:
The process cannot access the file 'N:\Dropbox\Apps\Logs\2016.12.08.log' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost)
   at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost)
   at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding)
   at System.IO.File.InternalAppendAllText(String path, String contents, Encoding encoding)
   at System.IO.File.AppendAllText(String path, String contents)
   at Mahou.Logging.Log(String logmsg, Int32 msgtype)
   at Mahou.KMHook.ReInitSnippets()
   at Mahou.MMain.Main(String[] args)
```

**Comment by BladeMight** [comment]
What do you do to achieve that?

**Comment by iamkarlson** [comment]
Just found that app is stopped and see that in log

**Comment by BladeMight** [comment]
OK, I'll try figure out why that happend.

**Comment by BladeMight** [comment]
Can you give me full log?

**Comment by iamkarlson** [comment]
Here you're
[2016.12.08.log.txt](https://github.com/BladeMight/Mahou/files/638683/2016.12.08.log.txt)

**Comment by BladeMight** [comment]
I tried a few times things you done to achieve error and get nothing... Can you reproduce it?

**Comment by BladeMight** [comment]
Here I added, logging bug detection, and logging off if 10+ **logging bugs** catched. [v1.4.3.7](https://github.com/BladeMight/Mahou/releases/tag/v1.4.3.7)

**Comment by avently** [comment]
Here is another log. 3-4 last versions are randomly crashes after some time. I do not no why.
[2016.12.07.txt](https://github.com/BladeMight/Mahou/files/640160/2016.12.07.txt)

**Comment by BladeMight** [comment]
@avently [that issue](https://github.com/BladeMight/Mahou/issues/37) in your log was fixed from [v1.4.3.6](https://github.com/BladeMight/Mahou/releases/tag/v1.4.3.6).
